cpp/sopt/l1_g_proximal.h

```diff
@@ -0,0 +1,140 @@
+  // parameter is required for internal logic in l1_proximal
+  L1GProximal(bool tight_frame = false)
+    : tight_frame_ (tight_frame),
+      l1_proximal_() {}
+  ~L1GProximal() {};
+
+// Implements the interface in GProximal
+
+  // Print log message with the correct norms
+  void log_message() const override {
+    SOPT_HIGH_LOG("Performing Forward Backward with L1 and L2 norms");
+  }
+
+  // Return the norm associated with this implementation
+  Real proximal_norm(t_Vector const &x) const override {
+    return sopt::l1_norm(static_cast<t_Vector>(Psi().adjoint() * x), l1_proximal_weights());
+  }
+
+  // Return g_proximal as a lambda function. Used in operator() in base class.
+  t_Proximal proximal_function() const override {
+    return [this](t_Vector &out, Real gamma, t_Vector const &x) {
+	     this -> l1_proximal(out, gamma, x);
+	   };
+  }
+
+  //! \brief Analysis operator Ψ
+  //! \details Under-the-hood, the object is actually owned by the l1 proximal.
+  t_LinearTransform const &Psi() const override {
+    return l1_proximal().Psi();
+  }
+
+
+// All the public properties below are specific to the l1 proximal
+// and therefore not part of the interface
+
+//! \brief L1 proximal used during calculation
+//! \details Non-const version to setup the object.
+  proximal::L1<Scalar> &l1_proximal() { return l1_proximal_; }
+  proximal::L1<Scalar> const &l1_proximal() const { return l1_proximal_; }
+  L1GProximal<SCALAR> &l1_proximal(proximal::L1<Scalar> const &arg) {
+    l1_proximal_ = arg;
+    return *this;
+  }
+
+// This macro creates get/setters that point to l1_proximal
+// In practice, we end up with a bunch of functions that make it simpler to set or get values
+// associated with the two proximal operators.
+// E.g.: `paddm.l1_proximal_itermax(100).l1_proximal_tolerance(1e-4)`.
+// ~~~
+#define SOPT_MACRO(VAR, TYPE)						 \
+  /** \brief Getter, forwards to l1_proximal **/                         \
+  TYPE const &l1_proximal_##VAR() const { return l1_proximal().VAR(); }  \
+  /** \brief Setter, forwards to l1_proximal **/                         \
+  L1GProximal<SCALAR> &l1_proximal_##VAR(TYPE const ARG) {               \
+    l1_proximal().VAR(ARG);				                 \
+    return *this;                                                        \
+  }
+
+  SOPT_MACRO(itermax, t_uint);
+  SOPT_MACRO(tolerance, Real);
+  SOPT_MACRO(positivity_constraint, bool);
+  SOPT_MACRO(real_constraint, bool);
+  SOPT_MACRO(nu, Real);
+  SOPT_MACRO(weights, t_Vector);
+#undef SOPT_MACRO
+
+  //! Analysis operator Ψ
+  template <class... ARGS>
+  typename std::enable_if<sizeof...(ARGS) >= 1, L1GProximal<SCALAR> &>::type Psi(
+      ARGS &&... args) {
+    l1_proximal().Psi(std::forward<ARGS>(args)...);
+    return *this;
+  }
+
+protected:
+
+  bool tight_frame_;
+  proximal::L1<Scalar> l1_proximal_;
+
+  // Helper functions for calling l1_proximal
+  //! Calls l1 proximal operator, checking for real constraints and tight frame
```

Comment:
```suggestion
  //! Calls l1 proximal operator, checking for real constraints
```
I think this bit only checks for the real constraint, tight frame is checked in the `call_l1_proximal`